doc/build/core/event.rst

```diff
@@ -140,6 +140,17 @@ this value can be supported::
     # it to use the return value
     listen(UserContact.phone, "set", validate_phone, retval=True)
 
+Multiprocessing Support
+-----------------------
+
+The event system supports multiprocessing in a limited manner.
```

Comment:
I disagree with the wording "in a limited manner", from my perspective it does not make any sense for events to "support multiprocessing", since event hooks pass live Python objects around.

This should be a simple ".. tip" section, and just mention, "note that when using Python multiprocessing or other approaches to create separate Python processes, events as they occur are only propagated within the same Python process in which they originated, there is no means by which an event would be transmitted to other running processes."

"In addition, establishment of event hooks themselves occurs locally to a Python process.   If a Python process forks a child process, existing event listeners will be copied to the child process at that point along with the rest of the Python interpreter state, but there is no intrinsic mechanism for event listener registration to propagate to other already existing processes."


will of course propagate from parent to child process, but an event hook established within a child process will not be present in the parent process, or in other sibling processes"

Reply:
Thank you @zzzeek for putting in some real thought into this, appreciated. I have incorporated all of this into the code. I made it a `tip` section.

Let me know if you think the `tip` should be moved to a separate section, or if it can stay in its own `Multiprocessing Support` section

Reply:
OK I think I misspoke when I used the phrase "tip section" , as this is now an entirely independent section with just a tip inside of it :)  

let's do this:

1. section is named "Events and Multiprocessing", where you have the section right now
2.  it's not a "tip" since this is a section of its own
3. text, will not start with "note that" since this is a first class section.   We also try to stick with more formal language outside of tutorials, so lets go with this

> SQLAlchemy's event hooks are implemented as Python data structures associated with a particular pair of Python functions and objects.   Event propagation itself is implemented as Python function calls.

> From this model, we can describe the behavior of the event system in terms of multiple Python processes, such as a parent child forking a child process, in the same way we would think about any other datastructure / function operation.     Event hooks registered in a parent process will be present in new child processes that are forked from that parent **after** the hooks have been registered, since the child process starts with a copy of all existing Python structures from the parent when it begins.  Child processes that already exist **before** the hooks are registered will **not** receive those new event hooks, as changes made to Python structures in a parent process do not propagate to child processes.

> For the events themselves, these are Python function calls, which do not have any ability to propagate between processes.  SQLAlchemy's event system does not implement any inter-process communication.    It is possible to implement event hooks that use Python inter-process messaging within them however this would need to be implemented by the user.

Reply:
Nice, thank you I like your changes and implemented them.  I did tweak the wording slightly as I felt it could be more concise, and have pushed the code. Feel free to take over this PR and directly commit any changes you think are necessary from here